sklearn/semi_supervised/label_propagation.py

```diff
@@ -287,6 +288,12 @@ def fit(self, X, y):
                     alpha, self.label_distributions_) + y_static
             remaining_iter -= 1
 
+        if remaining_iter <= 1:
```

Comment:
I think this is right given the current implementation, but the current implementation is buggy in that it seems you can never obtain n_iter_==max_iter.

Please fix the convergence condition and inline the code for _not_converged. I would then write something like:

```python
while self.n_iter_ < self.max_iter:
    if ... < tol:
        break
    Do stuff
    self.n_iter_ += 1
else:
    Warn
```